### DISCUSSION TITLE
can u provide a material-react-table standard  New and Update example with using @tanstack/react-query

### DISCUSSION THREAD

**Discussion by jjwallaby** [KevinVandy/material-react-table#964] [Q&A]
Hi,

can u provide a material-react-table standard  New and Update example with using @tanstack/react-query.

Thanks

**Comment by jjwallaby** [comment]
Trying to use the material-react-table without @tanstack/react-query.  for example of create new row and update.   just using the 
native react state management and not react-query

**Reply by KevinVandy** [reply]
I can't make an example for every possible combination of libraries with MRT. Some things you just have to figure out, though I have provided so many examples in my own docs, that you should be able to easily connect the dots to do what you want to do.

A reason why TanStack Query is so popular is that it drastically reduces the amount of code needed to do stuff like that with way less bugs, edge cases, and race conditions.

I already have a useEffect fetching example provided here: https://www.material-react-table.com/docs/examples/remote

Though be aware that if you write code like that, you'll probably have a lot of bugs.

I recommend reading this: https://tkdodo.eu/blog/why-you-want-react-query

**Comment by jjwallaby** [comment]
No problem Kevin,  Able to do easily.  Thanks for all the examples.

Understand using tanStack... 

eg.

  //CREATE action
  const handleCreateUser = async ( { values, row, table }) => {
    const newValidationErrors = validateRecord(values);
    console.log("VALID ERROR valudate CHECK:",newValidationErrors);
    if (Object.values(newValidationErrors).some((error) => error)) {
      console.log("VALID ERROR valudate YES:",newValidationErrors);
      setValidationErrors(newValidationErrors);
      return;
    }
    setValidationErrors({});
    // ADD row to table.
    let newdata=[];
    for(let no=0;no<data.length;no++) {
         let j_rec=data[no];
         newdata.push(j_rec);
    };
    newdata.push(values);
    setData(newdata);
    table.setCreatingRow(null); //exit creating mode
  };
